### PR TITLE
jax integration for for DefaultQubit2

### PR DESCRIPTION
**Pull request by timmysilv**
This follows #3903, but allows for execution with jax! I think this is easiest to review one commit at a time. They are:
1. copy-paste the autograd tests
2. change autograd to jax in that test file, update shots in pennylane source as needed
3. add the `argnums` attribute where appropriate to be jax-friendly

### CODE REVIEW

**Comment by timmysilv** [comment]
turning into a draft so we can factor the changes from #4118 into this PR before merging

File: doc/releases/changelog-dev.md
Comment (mudit2812) [inline doc/releases/changelog-dev.md:100]:
```suggestion
* The new device interface in integrated with `qml.execute` for Jax.
```

File: pennylane/interfaces/jax.py
Comment (mudit2812) [inline pennylane/interfaces/jax.py:468]: Shouldn't this have `tape.shots.shot_vector` if `new_device_interface==True`?

File: tests/interfaces/default_qubit_2_integration/test_jax_default_qubit_2.py
Comment (mudit2812) [inline tests/interfaces/default_qubit_2_integration/test_jax_default_qubit_2.py:1]: This looks good, but I'd suggest holding off until #4118 is merged, or use it as the base, so that the test file has the changes to autograd testing that Christina added in that PR

Comment (mudit2812) [reply tests/interfaces/default_qubit_2_integration/test_jax_default_qubit_2.py:1]: Just noticed you've already noted the aforementioned PR. 🚀

File: pennylane/interfaces/jax.py
Comment (timmysilv) [reply pennylane/interfaces/jax.py:468]: it's a batch of tapes so there is not (necessarily) a single value to pass. I instead handle this in the code this is calling on a tape-by-tape basis (if we get there and shots is None, we use `tape.shots.shot_vector` there

**Comment by timmysilv** [comment]
[sc-37944]

**Review by rmoyard — APPROVED** [review]
Thanks @timmysilv it looks good to me :100:

File: pennylane/interfaces/jax.py
Comment (timmysilv) [reply pennylane/interfaces/jax.py:468]: I've updated autograd now too, but there is still an assumption being made that if one tape has partitioned shots, they all do. what can ya do

**Review by mudit2812 — APPROVED** [review]

**Pull request merged**
